backend/src/order.ts

```diff
@@ -354,162 +336,252 @@ export function getAllForShow(show_id: number): Promise<IAdminOrderListItem[]> {
     {show_id: show_id}));
 }
 
-export function preparePayment(order_id: number): Promise<any> {
+async function getPaymentHandler(order_id: number): Promise<IPayment> {
+
+  let res: any, err: Error, conn: IConnection;
+
+  [conn, err] = await to(db.beginTransaction());
```

Comment:
You don't need a transaction for a single SELECT

Reply:
Ah, it was used also for the queries below. I don't think there's any need to do that UPDATE there? The SQL evolution should take care of that, and if the data doesn't exist, better not write anything there as we can't be sure it'd be correct. So I'd say remove the UPDATE and then you can also remove the transaction.

Reply:
Logic behind here is to set provider for order if not already presented. SQL evolution is taking care of all previously inserted orders. This is just for resolving new orders which do not have provider associated with them. 

Like:
order created -> trying to create payment -> no provider found -> assing this payment to our current provider in use -> now each payment has provider

Reply:
Ah, I see. Maybe do it at https://github.com/teekkarispeksi/nappikauppa2/pull/133/files#diff-73bf139ad02ab64435a54c3a9c58e7a6R410 instead? Then you could also remove the call to getPaymentProvider from https://github.com/teekkarispeksi/nappikauppa2/pull/133/files#diff-73bf139ad02ab64435a54c3a9c58e7a6R439 as it would be known already. 

I think the getter is wrong place for this.

Reply:
See `db/evolutions/02_.....`